File: app/src/main/java/com/example/marvelworld/ui/MarvelDetailViewModelFactory.java
package com.example.marvelworld.ui;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class MarvelDetailViewModelFactory implements ViewModelProvider.Factory {

    private String apiKey;
    private int id;

    public MarvelDetailViewModelFactory(String apiKey, int id) {
        this.apiKey = apiKey;
        this.id = id;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(MarvelDetailViewModel.class)) {
            return (T) new MarvelDetailViewModel(apiKey, id);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: app/src/main/java/com/example/marvelworld/ui/MarvelDetailViewModelFactory.kt
package com.example.marvelworld.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class MarvelDetailViewModelFactory(private val apiKey: String, private val id: Int) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(MarvelDetailViewModel::class.java)) {
            return MarvelDetailViewModel(apiKey, id) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}